### ISSUE TITLE
Reward / bonus extensions

### ISSUE THREAD

**Issue by DjWarmonger** [vcmi/vcmi#1752]
In 1.2 the ability to configure rewards in objects is expanded, however it still doesn't cover many fields VCMI already supports. Most of these seem easy to handle.

Extensions to json configuartions:

- [x] Banks get same reward features as configurable object (Limiters, Bonuses)
Needed to implement HoTA Ruins: https://github.com/orgs/vcmi/projects/3/views/1?pane=issue&itemId=24274704
- [ ] Random value of a Bonus (ie. grant hero 3 -7 mana point regeneration permanently)
- [x] Select **all** rewards that apply (ie. WoG Alms House)
- [ ] Grant global Bonus to Player
- [ ] Grant global Bonus to Team

More customisation for artifact types:

- [ ] Random spell scroll of specific level (or any level)
- [ ] Random combined artifact
- [ ] Random war machine (any or of specific rarity)
- [ ] Random Commander artifact (any or of specific rarity)
- [ ] Random Stack artifacts (any or of specific rarity)

Commander-specific rewards

- [ ] Grant Commander experience
- [ ] Grant any Bonus to Commander (WoG pickable gems)
- [ ] Grant Commander Skill
- [ ] Limiter for alive Commander

Stack-specific rewards

- [ ] Grant stack experience

Map Editor support

- [ ] Ability to configure all of the above for Pandora Box, Event and Seer Hut

**Comment by IvanSavenko** [comment]
> Banks can grant any reward from 1.2 extension 

More like banks should have shared code with configurable objects, or even converted to such

> More customisation for artifact types

Spell scroll should probably be its own reward class and not part of "artifacts" node.
For rest - perhaps introduce "traits" for artifacts? Either manually marked or autogenerated?


Also, if somebody starts work on this one or on #1751 - please note that somewhere, since right now we have me, @DjWarmonger and @rilian-la-te with some interest in these features.

**Comment by dydzio0614** [comment]
I would eventually want to see bonuses that can be configured in map or random generator template that affect all players at all time

**Comment by rilian-la-te** [comment]
@dydzio0614 now it is possible in a mod)

**Comment by rilian-la-te** [comment]
@DjWarmonger even if scout is slain?

**Comment by DjWarmonger** [comment]
@Alucard648 Yes, but that's a different feature altogether. I'm not sure if it's trivial to implement.

Also I can't think of any such objects ever used in mods, they were only rumored for Days of Reckoning. We may want to wait for these for now.

**Comment by IvanSavenko** [comment]
Closing
- Partially implemented in #5690 
- remaining points were moved to #5694
